### DISCUSSION TITLE
Dimensionality reduction

### DISCUSSION THREAD

**Discussion by jiyunshin** [dlopezg/mvpalab#13] [Q&A] [answered]
Hello David,

First, thanks so much for making and sharing MVPAlab. It's very well designed and easy to follow.

I have a theoretical question regarding dimensionality reduction and was hoping if I can get your advice.

I'm working on intracranial EEG data collected from multiple patients. As you might know, unlike EEG or MEG, the location of electrodes and the number of electrodes differ a lot across patients in iEEG.
My colleague recommended to do PCA for temporal generalization analysis but I'm not sure if it makes sense when I have different sets of electrodes and different numbers of electrodes.
Do you recommend to use dimensionality reduction in this case? What would be the disadvantage of not using dimensionality reduction? Will it affect the result or interpretation of the result?

Because of this idiosyncrasy, many researchers create a super-subject by pulling all electrodes but I also have different numbers of trials across subjects so will need to resample the trials and not sure if this is a best practice.

I will very much appreciate your advice here!

Many thanks,

Jiyun

**Comment by dlopezg** [comment, marked as answer]
Hi Jiyun,

Thank you for your positive feedback. Unfortunately, I do not have the required expertise in iEEG experimentation to recommend how to proceed in your situation. In EEG studies, one of the main disadvantages of using dimensionality reduction techniques, such as PCA, is that the spatial information of each electrode is lost. If you train your classifiers with raw data, each feature of your training set corresponds to the recorded voltage for each electrode. Thus, you can study how each of those features contributes to the decoding accuracy by analyzing the associated weight vector, for example. However, if you apply PCA, your data is projected onto a new coordinate system. As a result, the features in your training set may not directly correspond to the original EEG signal, and the interpretation of the classification results could be challenging. I hope this information is helpful.

Best,

David

**Reply by jiyunshin** [reply]
Thanks so much David. You comment on interpreting results after dimensionality reduction makes sense. I will think more if I need to know each electrode's contribution to decoding.

I have another slightly unrelated question.
I would like to do within-subject statistics for temporal generalization analysis and do you think my approach is correct?
So first I load only one subject data and set the cfg.stats.nper = 1000 and cfg.stats.nperg = 1000.
If I understand correctly, this will be equal to doing 1000 permutation tests within that subject?

Many thanks,

Jiyun

**Reply by dlopezg** [reply]
Hi Jiyun,

You can indeed compute within-subject statistics by simply loading data from a single subject and configuring the variables `cfg.stats.nper` and `cfg.stats.nperg` to the required number of permutations for your analysis. Please note that the number of permutations used for calculating all significant thresholds is always determined by `cfg.stats.nperg`, regardless of whether you are computing within-subject statistics or not. Therefore, in your case, `cfg.stats.nperg` should be at least equal to `cfg.stats.nper`.

Best regards,
David

**Reply by jiyunshin** [reply]
Thanks a lot David!
